tool/typed_mockito_test.dart

```diff
@@ -0,0 +1,497 @@
+      if (expectedMessage != e.message) {
+        throw new TestFailure("Failed, but with wrong message: ${e.message}");
+      }
+    }
+  }
+}
+
+void main() {
+  RealClass mock;
+
+  setUp(() {
+    mock = new MockedClass();
+  });
+
+  tearDown(() {
+    // In some of the tests that expect an Error to be thrown, Mockito's
+    // global state can become invalid. Reset it.
+    resetMockitoState();
+  });
+
+  group("spy", () {
+    setUp(() {
+      mock = spy(new MockedClass(), new RealClass());
+    });
+
+    test("should behave as mock when expectation are set", () {
+      when(mock.methodWithoutArgs()).thenReturn('Spied');
+      expect(mock.methodWithoutArgs(), 'Spied');
+    });
+  });
+
+
+  group("mixin support", () {
+    test("should work", () {
+      var foo = new MockFoo();
+      when(foo.baz()).thenReturn('baz');
+      expect(foo.bar(), 'baz');
+    });
+  });
+
+  group("when()", () {
+    test("should mock method without args", () {
+      when(mock.methodWithoutArgs()).thenReturn("A");
+      expect(mock.methodWithoutArgs(), equals("A"));
+    });
+    test("should mock method with normal args", () {
+      when(mock.methodWithNormalArgs(42)).thenReturn("Ultimate Answer");
+      expect(mock.methodWithNormalArgs(43), isNull);
+      expect(mock.methodWithNormalArgs(42), equals("Ultimate Answer"));
+    });
+    test("should mock method with mock args", () {
+      var m1 = new MockedClass();
+      when(mock.methodWithObjArgs(m1)).thenReturn("Ultimate Answer");
+      expect(mock.methodWithObjArgs(new MockedClass()), isNull);
+      expect(mock.methodWithObjArgs(m1), equals("Ultimate Answer"));
+    });
+    test("should mock method with positional args", () {
+      when(mock.methodWithPositionalArgs(42, 17)).thenReturn("Answer and...");
+      expect(mock.methodWithPositionalArgs(42), isNull);
+      expect(mock.methodWithPositionalArgs(42, 18), isNull);
+      expect(mock.methodWithPositionalArgs(42, 17), equals("Answer and..."));
+    });
+    test("should mock method with named args", () {
+      when(mock.methodWithNamedArgs(42, y: 17)).thenReturn("Why answer?");
+      expect(mock.methodWithNamedArgs(42), isNull);
+      expect(mock.methodWithNamedArgs(42, y: 18), isNull);
+      expect(mock.methodWithNamedArgs(42, y: 17), equals("Why answer?"));
+    });
+    test("should mock method with List args", () {
+      when(mock.methodWithListArgs([42])).thenReturn("Ultimate answer");
+      expect(mock.methodWithListArgs([43]), isNull);
+      expect(mock.methodWithListArgs([42]), equals("Ultimate answer"));
+    });
+    test("should mock getter", () {
+      when(mock.getter).thenReturn("A");
+      expect(mock.getter, equals("A"));
+    });
+    test("should mock hashCode", () {
+      named(mock, hashCode: 42);
+      expect(mock.hashCode, equals(42));
+    });
+    test("should have hashCode when it is not mocked", () {
+      expect(mock.hashCode, isNotNull);
+    });
+//    test("should n't mock toString", (){
```

Comment:
Phhh these files come with a lot of commented out code. Don't know what it's there for. Should I change it in `test/mockito_test.dart` and in `tool/typed_mockito_test.dart`?